github/GithubRetry.py

```diff
@@ -78,8 +80,8 @@ def __init__(self, secondary_rate_wait: float = DEFAULT_SECONDARY_RATE_WAIT, **k
         # 403 is too broad to be retried, but GitHub API signals rate limits via 403
         # we retry 403 and look into the response header via Retry.increment
         # to determine if we really retry that 403
-        kwargs["status_forcelist"] = kwargs.get("status_forcelist", list(range(500, 600))) + [403]
-        kwargs["allowed_methods"] = kwargs.get("allowed_methods", Retry.DEFAULT_ALLOWED_METHODS.union({"GET", "POST"}))
+        kwargs["status_forcelist"] = kwargs.get("status_forcelist", DEFAULT_STATUS_FORCELIST) + [403]
```

Comment:
```suggestion
        kwargs["status_forcelist"] = kwargs.get("status_forcelist", DEFAULT_STATUS_FORCELIST)
```

Reply:
Given a user provides a custom `status_forcelist` value without 403, the current logic will still add 403 on init. If we apply this change, this won't happen anymore and 403 won't be in the list. This could affect the custom 403 logic applied for GithubRetry. What do you think?

If you feel this is a non-issue, please go ahead and apply. I'll be offline for a week or so, and won't be able to check on this.